interlok-core/src/test/java/com/adaptris/core/services/splitter/XpathSplitterTest.java

```diff
@@ -167,18 +167,22 @@ public void testSplit() throws Exception {
       }
     }
     assertEquals(3, count);
+  }
 
-    // test with retainBranchNodes = true
-    splitter = new XpathMessageSplitter("/envelope/body/document");
-    splitter.setRetainBranchNodes(true);
-    count = 0;
-    msg = AdaptrisMessageFactory.getDefaultInstance().newMessage(XML_MESSAGE_WITH_SIBLINGS);
+  @Test
+  public void testSplitRetainBranchNodes() throws Exception {
+    AdaptrisMessage msg = AdaptrisMessageFactory.getDefaultInstance().newMessage(XML_MESSAGE_WITH_SIBLINGS);
+    String obj = "ABCDEFG";
     msg.addObjectHeader(obj, obj);
+    XpathMessageSplitter splitter = new XpathMessageSplitter("/envelope/body/document");
+    splitter.setRetainBranchNodes(true);
+    XmlUtils xml = new XmlUtils();
+    int count = 0;
     try (CloseableIterable<AdaptrisMessage> closeable = splitter.splitMessage(msg)) {
       for (AdaptrisMessage m : closeable) {
         assertFalse(m.getObjectHeaders().containsKey(obj));
         xml.setSource(new ByteArrayInputStream(m.getPayload()));
-        assertNotNull(xml.getSingleNode("/envelope/body/document"));
+        assertEquals(xml.getSingleNode("/envelope/body/document/text()").getNodeValue(), String.valueOf(count+1));
```

Comment:
It assert that the node exist but maybe we should check that the other nodes don't exist?